### ISSUE TITLE
forecasting: selecting exogeneous variables

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#1268]
I think there should be wrappers that allow to select which exogeneous variables to use, which is useful in cases where forecasters behave differently if they get exogeneous variabes or not. Notably, a large number of "garbage variables" can hamper performance, so selection of exogeneous variables may be helpful, and enables automated selection, too.
 For instance, useful wrappers could be:

* `IgnoreExogeneous` which blocks any `X` from being passed.
* `SelectExogeneous` which takes a list of column names, and only passes those ones through.

The `SelectExogeneous` can also be used in combination with tuning for automated feature selection.

**Comment by RNKuhns** [comment]
@fkiraly would the idea also be to eventually add functionality for automated exogenous variable selection methods?

**Comment by aiwalter** [comment]
I just wanted to open an issue when I saw this one :) so I rather write my idea here:
___

**Is your feature request related to a problem? Please describe.**
I would like to add a transformer that can selection features based on feature based on feature importance values of a meta-model/regressor.

Related: #1268
**Describe the solution you'd like**
```python
SelectExogeneous(method="feature_importances", regressor=DecisionTreeRegressor(), n_features=12, max_features=None, random_state=None)
```
where `regressor` is any `sklearn-like` tree-based model with an attribute `feature_importances_`. The regressor would be fitted on the `y` as endog and `X` as exog. The regressor could even be a `GridSearchCV` if user want to tune the meta-model.

Other methods that come to my mind:

- Random sampling method where features are sampled randomly. 

But I would start with the proposed solution.
**Additional context**
- pipelining
- tree-based models
- panel/global forecasting

@fkiraly do you agree that this class should be implemented as a `_SeriesToSeriesTransformer`? Bcz it will be part of tuning so it could be tuned in a `ForecastingPipeline`

**Comment by aiwalter** [comment]
I have already started implementing a transformer, but now I faced issues with the transformer interface. So the issue is, that the regressor would need `y` and `X` so also the transformer needs both then. But there is no such case in the transformer interface where `y` and `X` is given and `Xt` is returned. See STEP [here ](https://github.com/sktime/enhancement-proposals/blob/main/steps/01_forecasting_api/forecasting_pipeline.md) where we wrote down the cases.

But the order is switched. So the transformer would get the data as follows:
```python
FeatureSelection(...).fit_transform(Z=X, X=y)
```
This switch is needed as we transform `X` by selecting some columns (features) and return `X` also and `y` is just needed to build the regressor. This switch looks a bit weird maybe and also then we need a new tag for this, so in `ForecastingPipeline` we can have an if statement to check if `y` and `X` have to be switched. As currently the `ForecastingPipeline` just does `Transformer().fit_transform(Z=X)` and is not giving `y`. We already use the `skip-inverse-transform` tag, so it might be ok to add another into the `ForecastingPipeline`?

**Comment by aiwalter** [comment]
I discussed this with @mloning and he proposed to always switch arguments in `ForecastingPipeline` to have `fit(Z=X, X=y)` as we also give both arguments in `TransformedTargetForecaster.fit(Z=y, X=X)`. I agree to this and it makes sense to me.

**Comment by aiwalter** [comment]
closed by my PR #1347
